packages/compiler/core/util.ts

```diff
@@ -322,3 +322,102 @@ class CaseInsensitiveMap<T> extends Map<string, T> {
     return super.delete(key.toUpperCase());
   }
 }
+
+interface OrderedMapKey<K> {
+  key: K;
+}
+
+export class OrderedMap<K, V> implements Map<K, V> {
```

Comment:
Name is also maybe not the best, as the additional functionality it provide over `Map` is not ordering but renaming a key while keeping order.